pyesbulk/__init__.py

```diff
@@ -68,23 +101,58 @@ def put_template(es, name, body):
 
         Failure modes are raised as exceptions.
     """
+    assert name is not None and mapping_name is not None and body is not None
     retry = True
     retry_count = 0
     backoff = 1
     beg, end = time.time(), None
+    try:
+        body_ver = int(body["mappings"][mapping_name]["_meta"]["version"])
+    except KeyError as e:
+        raise Exception(
+            f"Bad template, {name}, given mapping name misssing from template,"
+            f" {mapping_name}: {e!r}"
+        )
     while retry:
+        try:
+            tmpl = es.indices.get_template(name=name)
+        except es_excs.ConnectionError:
+            # We retry all connection errors
+            _sleep_w_backoff(backoff)
+            backoff += 1
+            retry_count += 1
+            continue
+        except es_excs.TransportError as exc:
+            # Only retry on certain 500 errors
+            if exc.status_code != 404:
+                if exc.status_code not in [500, 503, 504]:
```

Comment:
Does Elasticsearch document certain 500 errors as retry-able? Normally, 500 errors are a sort of boggy nowhere since the server is telling you it doesn't know what happened.

Alternatively, since this is a GET operation, which had better be idempotent, there seems little harm in retrying an unknown error at least once.